apps/web/src/app/advertisers/campaigns/page.tsx

```diff
@@ -1,28 +1,163 @@
+import { Campaign, Network } from '@repo/types';
+import { set } from 'date-fns';
 import Link from 'next/link';
 import { useEffect, useState } from 'react';
 export default function CampaignsPage(): JSX.Element {
 	const [campaigns, setCampaigns] = useState<Campaign[]>([]);
+	const [cpms, setCpms] = useState<number[]>([]);
+	const [cpcs, setCPCs] = useState<number[]>([]);
+	const [ctrs, setCtrs] = useState<number[]>([]);
+	const [spends, setSpends] = useState<number[]>([]);
+	const [impressions, setImpressions] = useState<number[]>([]);
+	const [clicks, setClicks] = useState<number[]>([]);
+	const [networkNames, setNetworkNames] = useState<string[]>([]);
 	const advertiserId = 1;
+
 	useEffect(() => {
 		const fetchData = async () => {
-			const response = await fetch('/api/campaigns?advertiserId=1');
-			const jsonData = await response.json();
-			const campaigns = jsonData as Campaign[];
-			setCampaigns(campaigns);
+			const campaigns = await fetch(
+				`/api/campaigns?advertiserId=${advertiserId}`,
+			)
+				.then((res) => {
+					return res.json();
+				})
+				.then((data: Campaign[]) => {
+					return data;
+				});
+
+			const promises = campaigns.map(async (campaign) => {
+				const clicks = await fetch(
+					`/api/analytics/clicks?campaignId=${campaign.campaignId}`,
+				)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data: number) => {
+						return data;
+					});
+
+				const impressions = await fetch(
+					`/api/analytics/impressions?campaignId=${campaign.campaignId}`,
+				)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data: number) => {
+						return data;
+					});
+
+				const spend = await fetch(
+					`/api/analytics/spend?campaignId=${campaign.campaignId}`,
+				)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data: number) => {
+						return Math.round(100 * data) / 100;
+					});
+
+				const ctr = await fetch(
+					`/api/analytics/ctr?campaignId=${campaign.campaignId}`,
+				)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data: number) => {
+						return Math.round(100 * data) / 100;
+					});
+
+				const cpm = await fetch(
+					`/api/analytics/cpm?campaignId=${campaign.campaignId}`,
+				)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data: number) => {
+						return Math.round(100 * data) / 100;
+					});
+
+				const cpc = await fetch(
+					`/api/analytics/cpc?campaignId=${campaign.campaignId}`,
+				)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data: number) => {
+						return Math.round(100 * data) / 100;
+					});
+
+				const networkName = await fetch(`/api/networks?networkId=2`)
+					.then((res) => {
+						return res.json();
+					})
+					.then((data) => {
+						const networks: Network[] = data.data;
+						if (!networks || networks.length === 0) {
+							return 'Fail';
+						}
+
+						return networks[0]!.name!;
+					});
```

Comment:
Right now these are happening sequentially. But I do see how you were trying to send them in parallel. What you need to do is remove the `await`s from the fetch requests which will turn everything into `Promise` objects. Then after you start all the `Promise`s, that's when you do the `await Promise.all([...])` where the arg is the list of `Promise`s you started. 

You can even do this one level higher too with every campaign to be mega parallel.